### ISSUE TITLE
Whitelist temporarily

### ISSUE THREAD

**Issue by jacklul** [badsgahhl/pihole-browser-extension#39]
How hard would it be to implement third button that whitelist the current URL for the x amount of seconds specified in the time field?

Sometimes you end up clicking a link that is a redirect through some kind of tracking/ad site and you just have no other choice but go through it just this time without permanently whitelisting it.

It could simply add record to whitelist and remove it after x seconds or use the clients & groups to whitelist it only for the current device IP (this would be probably preferred). I have no idea if required API methods are available to do it the second way.

**Comment by badsgahhl** [comment]
Lovely idea but impossible due to the lack of api endpoints of PiHole. The PiHole Api only can add/remove domains either from Whitelist/Blacklist or with the Whitelist Regex/Blacklist Feature what you also have in the browser.
Every other feature is only available in the Web Interface.

I'm sorry, closing this here :(

**Comment by jacklul** [comment]
Well I thought you could just send whitelist request, wait 10 seconds and send remove from whitelist.
Guess it's not supported in addons?

**Comment by badsgahhl** [comment]
Oh sorry I looked at the groups & clients part.

Yes this could work. I will check out how it fits into the extension.

**Comment by amikofalvy** [comment]
Is this repository still active? If I submit a PR for this feature will it get reviewed and published?

**Comment by badsgahhl** [comment]
Yes this project is active. Personally I currently don't add any new features, as of personal time limitations and the fact that I don't use PiHole that often anymore.

Feel free to create a PR, I will definitely look into it.

**Comment by HyperCriSiS** [comment]
What are you using now instead of pihole?

**Comment by badsgahhl** [comment]
> What are you using now instead of pihole?

Mainly Adguard Home. Just because "it works". PiHole sometimes killed my raspberry or it had other issues. Adguard Home was then just an one to one replacement.

**Comment by HyperCriSiS** [comment]
> > What are you using now instead of pihole?
> 
> Mainly Adguard Home. Just because "it works". PiHole sometimes killed my raspberry or it had other issues. Adguard Home was then just an one to one replacement.

Does it also work well with unbound?
This also happend to me, but it seems it was always the micro SD, all of them suck really fast.
Does everything else work the same or better than with pihole?

**Comment by badsgahhl** [comment]
> > > What are you using now instead of pihole?
> > 
> > 
> > Mainly Adguard Home. Just because "it works". PiHole sometimes killed my raspberry or it had other issues. Adguard Home was then just an one to one replacement.
> 
> Does it also work well with unbound? This also happend to me, but it seems it was always the micro SD, all of them suck really fast. Does everything else work the same or better than with pihole?

Didn't use unbound so I cannot give any advice or info about that with Adguard. Everything else worked for fluently like I knew it from PiHole, but I only used basic features of PiHole and not special ones.
